tests/protocol/test_connection.py

```diff
@@ -26,3 +26,93 @@ def test_kill(self):
     def test_ping(self):
         self.send_request("ping")
         self.assertEqualResponse("OK")
+
+    def test_tagtypes(self):
+        self.send_request("tagtypes")
+        self.assertInResponse("tagtype: Artist")
+        self.assertInResponse("tagtype: ArtistSort")
+        self.assertInResponse("tagtype: Album")
+        self.assertInResponse("tagtype: AlbumArtist")
+        self.assertInResponse("tagtype: AlbumArtistSort")
+        self.assertInResponse("tagtype: Title")
+        self.assertInResponse("tagtype: Track")
+        self.assertInResponse("tagtype: Name")
+        self.assertInResponse("tagtype: Genre")
+        self.assertInResponse("tagtype: Date")
+        self.assertInResponse("tagtype: Composer")
+        self.assertInResponse("tagtype: Performer")
+        self.assertInResponse("tagtype: Disc")
```

Comment:
You added "Comment", do you want "Comment" here?

Reply:
I added comment because I saw that it was being used somewhere. I probably should add all of the tagtypes listed here https://mpd.readthedocs.io/en/latest/protocol.html#tags. There seem to be some additional tagtypes being used that are not in that list though (i.e. X-AlbumUri) so I'm not sure how to handle those.

Reply:
I'm not sure about that. tagtypes used to be live in the "reflection" area of MPD protocol because clients are supposed to only use tags reported by that command. So we should probably only report tags we support. Listing the others doesn't have any benefit. Don't worry about our Mopidy-specific "X-something" tags, they are fine as they are but anything else we should probably stop including in our responses..

Reply:
Should the Mopidy-specific tags be added to the tagtypes list? What about other tags like "file", "Pos" (Position?), and "Id"? Should those be added to the tagtypes list as well, and therefore be something that clients can specify they are/aren't interested in, or should those still be treated separately? Also just to clarify I was going to change the tagtype list to only include the tags being used in the translator.py track_to_mpd_format function, was that what you meant by only including tags we actually support?

Reply:
The tagtypes stuff is specifically related to track *metadata* fields. It shows clients what metadata tags the server understands so they avoid using unknown ones in their command arguments. It also clients to reduce response output they don't care about. Fields like `id` and `pos` are not metadata, they are essentially part of the MPD protocol and all servers will support them. Arguably a client also might not care about some of these but in general there's not much you can do without them so disabling them is a weird use-case. Plus I don't believe mpd supports disabling them so neither will we. So `id`, `pos` etc should continue not to be part of the tagtypes list.

In general, the Mopidy-specific `X-` tags are metadata so they should stay part of the tagtypes list (and thus included in the tests here). However, `X-AlbumImage` is no longer used anywhere in Mopidy-MPD so we could drop that one.

> I was going to change the tagtype list to only include the tags being used in the translator.py track_to_mpd_format function, was that what you meant by only including tags we actually support?

The tagtype list should be a superset of:
* all track metadata-like fields we output in the translator
* all tag fields we support passed in command arguments.

The latter means adding `comment` to the list is a good idea.

Reply:
Ok I think the tagtype list should now be reduced to only those actually used by mopidy-mpd.